src/catkin_pkg/package.py

```diff
@@ -696,7 +696,25 @@ def parse_package_string(data, filename=None, warnings=None):
         })
     nodes = [n for n in root.childNodes if n.nodeType == n.ELEMENT_NODE]
     unknown_tags = set([n.tagName for n in nodes if n.tagName not in known.keys()])
-    if unknown_tags:
+    if unknown_tags == set(['run_depend']) and pkg.package_format >= 2:
+        errors.append('\033[91mWARNING:')
+        errors.append('WARNING: ACTION REQUIRED')
+        errors.append('WARNING:')
+        errors.append('WARNING: The manifest of package "%s" (with format version %d) '
+                      'does not support <run_depend>. ' % (pkg.name, pkg.package_format))
+        errors.append('WARNING: Please update your "%s/package.xml" with <exec_depend>' % (pkg.name))
+        errors.append('WARNING:')
+        errors.append('WARNING:\033[0m')
+    elif unknown_tags == set(['exec_depend']) and pkg.package_format < 2:
+        errors.append('\033[91mWARNING:')
+        errors.append('WARNING: ACTION REQUIRED')
+        errors.append('WARNING:')
+        errors.append('WARNING: The manifest of package "%s" (with format version %d) '
+                      'does not support <exec_depend>. ' % (pkg.name, pkg.package_format))
+        errors.append('WARNING: Please update your "%s/package.xml" with <run_depend>' % (pkg.name))
+        errors.append('WARNING:')
+        errors.append('WARNING:\033[0m')
+    elif unknown_tags:
```

Comment:
I would suggest to always print this message first. And conditionally the more detailed messages afterwards providing specific suggestions for run/exec dependencies.